src/syscheckd/db/src/fimDB.cpp

```diff
@@ -0,0 +1,257 @@
+#ifdef WIN32
+        m_rsyncHandler->startSync(m_dbsyncHandler->handle(), nlohmann::json::parse(FIM_REGISTRY_START_CONFIG_STATEMENT), m_syncMessageFunction);
+        m_rsyncHandler->startSync(m_dbsyncHandler->handle(), nlohmann::json::parse(FIM_VALUE_START_CONFIG_STATEMENT), m_syncMessageFunction);
+#endif
+        m_loggingFunction(LOG_INFO, "Finished FIM sync.");
+    }
+    catch (const std::exception& ex)
+    {
+        m_loggingFunction(LOG_ERROR, ex.what());
+    }
+}
+
+void FIMDB::loopRSync(std::unique_lock<std::mutex>& lock)
+{
+    m_loggingFunction(LOG_INFO, "FIM sync module started.");
+    sync();
+
+    while (!m_cv.wait_for(lock, std::chrono::seconds{m_interval_synchronization}, [&]()
+{
+    return m_stopping;
+}))
+    {
+        sync();
+    }
+    m_rsyncHandler.reset(nullptr);
+}
+
+#ifdef WIN32
+void FIMDB::init(unsigned int interval_synchronization,
+                 unsigned int max_rows_file,
+                 unsigned int max_rows_registry,
+                 send_data_callback_t callbackSync,
+                 logging_callback_t callbackLog,
+                 std::unique_ptr<DBSync> dbsyncHandler,
+                 std::unique_ptr<RemoteSync> rsyncHanlder)
+#else
+void FIMDB::init(unsigned int interval_synchronization,
+                 unsigned int max_rows_file,
+                 send_data_callback_t callbackSync,
+                 logging_callback_t callbackLog,
+                 std::unique_ptr<DBSync> dbsyncHandler,
+                 std::unique_ptr<RemoteSync> rsyncHanlder)
+#endif
+{
+    // LCOV_EXCL_START
+    std::function<void(const std::string&)> callbackSyncWrapper
+    {
+        [callbackSync](const std::string & msg)
+        {
+            callbackSync(FIM_LOCATION, msg.c_str());
+        }
+    };
+    // LCOV_EXCL_STOP
+
+    std::function<void(modules_log_level_t, const std::string&)> callbackLogWrapper
+    {
+        [callbackLog](modules_log_level_t level, const std::string & log)
+        {
+            callbackLog(level, log.c_str());
+        }
+    };
+
+    m_interval_synchronization = interval_synchronization;
+    m_max_rows_file = max_rows_file;
+#ifdef WIN32
+    m_max_rows_registry = max_rows_registry;
+#endif
+    m_dbsyncHandler = std::move(dbsyncHandler);
+    m_rsyncHandler = std::move(rsyncHanlder);
+    m_syncMessageFunction = callbackSyncWrapper;
+    m_loggingFunction = callbackLogWrapper;
+    m_stopping = false;
+
+    setFileLimit();
+#ifdef WIN32
+    setRegistryLimit();
+    setValueLimit();
+#endif
+
+}
+
+int FIMDB::insertItem(const nlohmann::json& item)
+{
+    try
+    {
+        m_dbsyncHandler->insertData(item);
+    }
+    catch (const DbSync::max_rows_error& ex)
+    {
+        m_loggingFunction(LOG_INFO, ex.what());
+        return 1;
+    }
+    catch (const std::exception& ex)
+    {
+        m_loggingFunction(LOG_ERROR, ex.what());
+        return 2;
+    }
+
+    return 0;
```

Comment:
Could you create a new const value with this returns? For instance: 

const int SUCESS = 0;

Reply:
Done!